contract_manager/src/executor.ts

```diff
@@ -12,7 +13,14 @@ export async function executeVaa(senderPrivateKey: PrivateKey, vaa: Buffer) {
   const parsedVaa = parseVaa(vaa);
   const action = decodeGovernancePayload(parsedVaa.payload);
   if (!action) return; //TODO: handle other actions
-  for (const contract of Object.values(DefaultStore.contracts)) {
+
+  let contracts:
+    | Record<string, PriceFeedContract>
+    | Record<string, EvmEntropyContract>;
+  if (action instanceof EvmExecute) contracts = DefaultStore.entropy_contracts;
```

Comment:
for EvmExecute action we don't need to loop over the contracts. We find the EvmChain via targetChainId and create a EvmExecutor class with the specified chain and `action.executorAddress` and then run the vaa on it.